localstorage/firebase-worker.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Dedicated web worker that loads Firebase off of the main browser thread.
+ * Each tab will run an instance of Firestore via the web worker. The worker
+ * receives messages from your main page and performs Firestore operations.
+ * 
+ * Must be loaded and initialized from your main page:
+ * dedicatedWorker = new Worker("firebase-worker.js");
+ */
+
+importScripts('../firebase/firebase-app.js');
+importScripts('../firebase/firebase-firestore.js');
+
+var firebaseConfig = {
+  apiKey: "api-key",
+  authDomain: "project-id.firebaseapp.com",
+  databaseURL: "https://project-id.firebaseio.com",
+  projectId: "project-id",
+  storageBucket: "project-id.appspot.com",
+  messagingSenderId: "sender-id",
+  appId: "app-id",
+  measurementId: "G-measurement-id",
+};
+
+/** The Firestore instance. */
+self.db = null;
+self.onmessage = async function(e) {
+  // Do Firebase work here.
+  if (e.data === 'persistenceAvailable') {
+    await initializeApp(true);
+  } else if (e.data === 'persistenceUnavailable') {
+    await initializeApp(false);
+  } else if (e.data === 'enableNetwork') {
```

Comment:
Perhaps to make it clearer that these are examples that could be removed / extended you should move the "do firebase work here" comment here and say:

// Add message handlers that do Firebase work here...

FWIW, I guess enabling / disabling the network is a very _simple_ example, but it's not something people would typically do in a real app...  Could we make our example do a simple write to the backend or something instead?